File: x-baseproject/src/main/java/com/xia/baseproject/ui/dialog/LoadingDialog.java
package com.xia.baseproject.ui.dialog;

import android.content.Context;
import android.support.annotation.NonNull;
import android.support.v7.widget.AppCompatTextView;
import android.text.TextUtils;
import android.view.Gravity;
import android.view.View;
import android.view.Window;

import com.wang.avi.AVLoadingIndicatorView;
import com.xia.baseproject.R;
import com.xia.baseproject.R2;
import com.xia.baseproject.ui.dialog.load.LoaderCreator;
import com.xia.baseproject.ui.dialog.load.LoaderStyles;

import butterknife.BindView;

/**
 * @author xia
 * @date 2018/7/29.
 */
@SuppressWarnings("WeakerAccess")
public class LoadingDialog extends BaseDialog {
    @BindView(R2.id.dialog_loading_iv)
    AVLoadingIndicatorView mAVLoadingIndicatorView;
    @BindView(R2.id.dialog_loading_show_tv)
    AppCompatTextView mLoadingTv;

    private static final String DEFAULT_LOADER = LoaderStyles.LineSpinFadeLoaderIndicator;

    private String mMessage;

    public LoadingDialog(@NonNull Context context) {
        this(context, "");
    }

    public LoadingDialog(@NonNull Context context, String message) {
        this(context, DEFAULT_LOADER, message);
    }

    public LoadingDialog(@NonNull Context context, @LoaderStyles.LoaderStyle String type, String message) {
        super(context);
        this.mMessage = message;
        LoaderCreator.create(type, mAVLoadingIndicatorView);
    }

    @Override
    protected int getLayoutId() {
        return R.layout.base_dialog_loading;
    }

    @Override
    protected void initDialog() {
        final Window window = getWindow();
        if (window != null) {
            window.setGravity(Gravity.CENTER);
            window.setDimAmount(0.2f);
            setCanceledOnTouchOutside(false);
        }
    }

    @Override
    public void show() {
        super.show();
        if (!TextUtils.isEmpty(mMessage)) {
            mLoadingTv.setText(mMessage);
            mLoadingTv.setVisibility(View.VISIBLE);
        }
        mAVLoadingIndicatorView.show();
    }

    @Override
    public void cancel() {
        super.cancel();
        mAVLoadingIndicatorView.hide();
    }
}

File: fly/src/main/java/com/xia/fly/ui/dialog/LoadingDialog.kt
package com.xia.fly.ui.dialog

import android.content.Context
import android.text.TextUtils
import android.view.Gravity
import android.view.View
import androidx.appcompat.widget.AppCompatTextView
import com.wang.avi.AVLoadingIndicatorView
import com.xia.fly.R
import com.xia.fly.ui.dialog.loader.LoaderCreator
import com.xia.fly.ui.dialog.loader.LoaderStyles

/**
 * @author xia
 * @date 2018/7/29.
 */
class LoadingDialog constructor(context: Context) : FlySupportDialog(context) {
    private lateinit var mAVLoadingIndicatorView: AVLoadingIndicatorView
    private lateinit var mLoadingTv: AppCompatTextView

    override fun getLayoutId(): Int {
        return R.layout.base_dialog_loading
    }

    override fun onBindAny(view: View) {
    }

    override fun initDialog() {
        setCanceledOnTouchOutside(false)
        window?.apply {
            setGravity(Gravity.CENTER)
            setDimAmount(0.2f)
        }

        mAVLoadingIndicatorView = findViewById(R.id.dialog_loading_iv)
        mLoadingTv = findViewById(R.id.dialog_loading_show_tv)
    }

    fun setMessage(message: String?) {
        if (!TextUtils.isEmpty(message)) {
            mLoadingTv.text = message
            mLoadingTv.visibility = View.VISIBLE
        }
    }

    fun setType(@LoaderStyles.LoaderStyle type: String?) {
        val typeTemp = type ?: DEFAULT_LOADER
        LoaderCreator.create(typeTemp, mAVLoadingIndicatorView)
    }

    override fun show() {
        mAVLoadingIndicatorView.smoothToShow()
        super.show()
    }

    override fun cancel() {
        mAVLoadingIndicatorView.smoothToHide()
        super.cancel()
    }

    companion object {

        private const val DEFAULT_LOADER = LoaderStyles.LINE_SPIN_FADE_LOADER_INDICATOR
    }
}
